sdk/python/kfp/cli/component_test.py

```diff
@@ -579,27 +578,28 @@ def test_existing_dockerfile_can_be_overwritten(self):
                 COPY . .
                 '''))
 
-    @unittest.skip(
-        "Skipping this test as it's failing. Refer to https://github.com/kubeflow/pipelines/issues/11038"
-    )
     def test_dockerfile_can_contain_custom_kfp_package(self):
         component = _make_component(
             func_name='train', target_image='custom-image')
         _write_components('components.py', component)
         package_dir = os.path.dirname(os.path.dirname(self.current_dir))
 
-        # suppresses large stdout from subprocess that builds kfp package
-        with mock.patch.object(
```

Comment:
I think it is actually the intention to mock subprocess runs, as unit tests should generally be isolated and dependency-free. Subprocesses introduce external dependencies that can make tests flaky and harder to debug.
Why was the mock not working?

Reply:
+1

Reply:
The [original error message](https://github.com/kubeflow/pipelines/issues/11038) indicates that the test can't find the built KFP wheel file in the temp directory.
This is because the mock of `subprocess.run` doesn’t perform the actual build process (e.g., running `python setup.py bdist_wheel`), it doesn't create the wheel file.
While the mock ensures `subprocess.run` doesn't actually execute, it fails to account for the necessary file outputs (i.e., `.whl` file). 
In this case mock wasn't building out the wheel file entirely. Replacing by `subprocess.run` did it.